tests/test-cmd/test-build/test.build.js

```diff
@@ -0,0 +1,103 @@
+import path from 'path';
+import {assert} from 'chai';
+
+import {safeFileName} from '../../../src/cmd/build';
+import {onlyInstancesOf, InvalidManifest} from '../../../src/errors';
+import {withTempDir, ZipFile} from '../../helpers';
+import * as fs from '../../../src/util/promised-fs';
+import {getPackageBasename, prepareBuildDir} from '../../../src/cmd/build';
+
+import * as adapter from './adapter';
+
+
+describe('build', () => {
+
+  describe('build', () => {
+
+    it('zips a package', () => {
+      let zipFile = new ZipFile();
+
+      return withTempDir(
+        (tmpDir) =>
+          adapter.buildMinimalExt(tmpDir)
```

Comment:
Proposal: How about introducing [co](https://github.com/tj/co) (at least in the test cases, because, in my own opinion, currently the implementation is clear enough even with the use of plain Promises) to get nicer async code?

as an example this test case could be turned into something like:
```js
it('zips a package', () => withTempDir(
  (tmpDir) =>
     co(function*() {
       let zipFile = new ZipFile();
 
       let {extensionPath} = yield adapter.buildMinimalExt(tmpDir);
 
       assert.match(extensionPath, /minimal_extension-1\.0\.xpi$/);
 
       yield zipFile.open(extensionPath);
 
       let fileNames = [];
 
       yield zipFile.readEach((entry) => fileNames.push(entry.fileName));
 
       assert.deepEqual(fileNames, ['manifest.json']);
     })
));
```

and as a plus: this mostly matches to our current coding style for async tests in the Firefox source code (e.g. what is achieved in the test suites using [add_task](https://dxr.mozilla.org/mozilla-central/source/browser/components/extensions/test/browser/browser_ext_browserAction_popup.js#129) and [Task.async](https://dxr.mozilla.org/mozilla-central/source/browser/components/extensions/test/browser/browser_ext_browserAction_popup.js#110)